packages/indexer-common/src/allocations/query-fees.ts

```diff
@@ -367,14 +410,177 @@ export class AllocationReceiptCollector implements ReceiptCollector {
+              allocation: allocations.find(
+                (a) => a.id === toAddress(signedRav.rav.allocationId),
+              ),
+              sender: rav.senderAddress,
+            }
+          })
+          .filter((rav) => rav.allocation !== undefined) as RavWithAllocation[] // this is safe because we filter out undefined allocations
+      },
+      { onError: (err) => this.logger.error(`Failed to query pending RAVs`, { err }) },
+    )
+  }
+
+  protected getSignedRAVsEventual(
+    pendingRAVs: Eventual<RavWithAllocation[]>,
+  ): Eventual<ValidRavs> {
+    return pendingRAVs.tryMap(
+      async (pendingRAVs) => {
+        return await pReduce(
+          pendingRAVs,
+          async (results, rav) => {
+            // allocationId doesn't have 0x prefix, but still matched in contract calls
+            // destory voucher if AllocationIDTracker.isAllocationIdUsed(rav.allocationId)
+            if (
+              BigNumber.from(rav.rav.rav.valueAggregate).lt(
+                this.voucherRedemptionThreshold,
+              )
+            ) {
+              results.belowThreshold.push(rav)
+            } else {
+              results.eligible.push(rav)
+            }
+            return results
+          },
+          { belowThreshold: <RavWithAllocation[]>[], eligible: <RavWithAllocation[]>[] },
+        )
+      },
+      { onError: (err) => this.logger.error(`Failed to reduce to signed RAVs`, { err }) },
+    )
+  }
+
+  // redeem only if last is true
+  // Later can add order and limit
+  protected async pendingRAVs(): Promise<ReceiptAggregateVoucher[]> {
+    let ravsToRedeem: ReceiptAggregateVoucher[] = []
+    const unfinalizedRAVs = await this.models.receiptAggregateVouchers.findAll({
+      where: { last: true, final: false },
+    })
+    // Obtain allocationIds to use as filter in subgraph
+    const allocationIdsAddress = unfinalizedRAVs.map(
+      (rav) => rav.getSignedRAV().rav.allocationId.toLowerCase(),
+    )
+    const allocationIds = unfinalizedRAVs.map(
+      (rav) => rav.getSignedRAV().rav.allocationId.toLowerCase().replace('0x', ''),
+    )
+
+    if (allocationIdsAddress.length > 0) {
+      console.log(`Obtaining redeemed RAVs against subgraph`)
+      const redeemedRavsQueryResp = await this.tapSubgraph.query(
+        gql`
+          query transactions($allocationIds: [String!]!) {
+            transactions(
+              where: { type: "redeem", allocationID_in: $allocationIds }
+            ) {
+              allocationID
+            }
+          }
+        `,
+        { allocationIdsAddress },
+      )
+      // From obtained allocationIDs, get those that were not returned and redeem
+      const returnedAllocationIDs = redeemedRavsQueryResp.data.transactions.map(
+        (transaction) => transaction.allocationID,
+      )
+      const nonRedeemedAllocationIDs = allocationIdsAddress.filter(
+        (allocationID) => !returnedAllocationIDs.includes(allocationID),
+      )
```

Comment:
How about resetting `redeemedAt` to `null` for all the `nonRedeemedAllocationIDs`. Just to make sure that the logic is solid in case of a reorg that reverts the redeem?

Reply:
Sounds good withe me yes